features/step_definitions/main_structure_steps.rb

```diff
@@ -29,13 +29,17 @@
 end
 
 Then (/^I login with username "([^\"]*)" and password "([^\"]*)"$/) do |username, password|
-  wait_for_element_exists("* id:'edit_password'", timeout: 6000)
+  step("I login with username \"%s\" and password \"%s\", without waiting for home screen" % [username, password])
+  wait_for_element_exists("* id:'home_gridview_buttons'", timeout: 60)
```

Comment:
the timeouts here are actually seconds, not milliseconds, so 6000 seemed like overkill. 6 seemed a little low though so settled on a minute for now. could probably still be lower but wanted to be conservative.